static/sharing.js

```diff
@@ -107,74 +107,151 @@ function updateShares(container, url) {
-                } else {
-                    var result = {
-                        updateState: updateState,
-                        extra: extra,
-                        url: newUrl,
-                    };
-                    if (result.updateState) {
-                        noteNewState(config, result.extra);
+            var cacheLinkId = currentBind;
+            if (currentBind === 'Embed') {
+                cacheLinkId = getEmbeddedCacheLinkId();
+            }
+            if (!urls[cacheLinkId]) {
+                label.text(currentNode.text());
+                permalink.val('');
+                getLinks(config, currentBind, function (error, newUrl, extra, updateState) {
+                    if (error || !newUrl) {
+                        permalink.prop('disabled', true);
+                        permalink.val(error || 'Error providing URL');
+                    } else {
+                        urls[cacheLinkId] = {
+                            updateState: updateState,
+                            extra: extra,
+                            url: newUrl,
+                        };
+                        onUpdate(socialSharing, config, currentBind, urls[cacheLinkId]);
                     }
-                    permalink.val(result.url);
-                }
-            });
+                });
+            } else {
+                onUpdate(socialSharing, config, currentBind, urls[cacheLinkId]);
+            }
+        }
+
+        root.find('.sources button').on('click', function () {
+            setCurrent($(this));
+            update();
         });
 
-        // Dismiss the popover on escape.
-        $(document).on('keyup.editable', function (e) {
-            if (e.which === 27) {
-                domLink.popover('hide');
-            }
+        var embeddedButton = $('.shareembed');
+        embeddedButton.on('click', function () {
+            setCurrent(embeddedButton);
+            update();
+            getLink.popover('hide');
         });
 
-        // Dismiss on any click that isn't either in the opening element, inside
-        // the popover or on any alert
-        $(document).on('mouseup', function (e) {
-            var target = $(e.target);
-            if (!target.is(domLink) && domLink.has(target).length === 0 && target.closest('.popover').length === 0)
-                domLink.popover('hide');
+        $('#embedsettings input').off('click').on('click', function () {
+            setCurrent(embeddedButton);
+            update();
         });
-    }
 
-    init(fullDomLink, 'Full');
-    init(shortDomLink, 'Short');
+        update();
+    }).prop('title', title);
+
+    // Dismiss the popover on escape.
+    $(document).on('keyup.editable', function (e) {
+        if (e.which === 27) {
+            getLink.popover('hide');
+        }
+    });
 
-    // Opens the full link popup if asked to by the editor
-    layout.eventHub.on('displaySharingPopover', function () {
```

Comment:
We'll need a solution for this - it's emitted on ctrl-S if "save" global option is turned off. I'm not sure what to do, we could:

* Always pop up the short url
* Pop up the selector dropdown (though then the user needs to press down and enter, which could be super confusing if they weren't expecint it)
* Or remember the last one the user did and bring that up (with opening the selector dropdown if not)
* Disable this and remoev the functionality in the settings.

I'm mildly in favour of the first one, just use a short share link. or the "remember" version though it's more work.

Reply:
reimplemented this with the triggering of creating a short url